cli/azd/pkg/project/scaffold_gen_util.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package project
+
+import (
+	"strings"
+	"unicode"
+)
+
+type location struct {
+	start int
+	stop  int
+}
+
+// parseEnvSubstVariables parses the envsubst expression(s) present in a string.
+// substitutions, returning the locations of the expressions and the names of the variables.
+//
+// It works with both:
+//   - ${var} and
+//   - ${var:=default} syntaxes
+func parseEnvSubstVariables(s string) (names []string, expressions []location) {
```

Comment:
We may at some point want a way to escape `${` in these strings.

Reply:
Indeed. I suspect that we'd perhaps also move away from `${VARIABLE}` into something more powerful for different kinds of use-cases.

For alpha, the current thinking is to expose the basic functionality enough for users to find the value in it.